File: typescript/src/address/search/searchSuggestion.ts
import { RestApiSearchResultSuggestion } from "../../server/address/search/restApiSearchResultSuggestion";
import { AdditionalAttribute } from "./additionalAttribute";

export type SearchSuggestion = {
    globalAddressKey: string;
    text: string;
    matched: number[][];
    formatUrl: string;
    dataset: string;
    additionalAttributes: AdditionalAttribute[]
}

export function restApiResponseToSuggestion(response: RestApiSearchResultSuggestion): SearchSuggestion {
    return {
        globalAddressKey: response.global_address_key??"",
        text: response.text??"",
        matched: response.matched??[],
        formatUrl: response.format??"",
        dataset: response.dataset??"",
        additionalAttributes: response.additional_attributes ? response.additional_attributes.map(attr => ({name: attr.name, value: attr.value})) : []

    }
}

File: csharp/DVSClient/DVSClient/address/search/SearchSuggestion.cs
using DVSClient.Server.Address.Search;

namespace DVSClient.Address.Search
{
    public class SearchSuggestion
    {
        public string GlobalAddressKey { get; }
        public string Text { get; }
        public IEnumerable<IEnumerable<int>> Matched { get; }
        public string FormatUrl { get; }
        public string Dataset { get; }
        public IEnumerable<AdditionalAttribute> AdditionalAttributes { get; }

        public SearchSuggestion(RestApiSearchResultSuggestion suggestion)
        {
            GlobalAddressKey = suggestion.GlobalAddressKey ?? string.Empty;
            Text = suggestion.Text ?? string.Empty;
            Matched = suggestion.Matched ?? new List<IEnumerable<int>>();
            FormatUrl = suggestion.FormatUrl ?? string.Empty;
            Dataset = suggestion.Dataset ?? string.Empty;
            AdditionalAttributes = suggestion.AdditionalAttributes != null
                ? suggestion.AdditionalAttributes.Select(attr => new AdditionalAttribute(attr)).ToList()
                : new List<AdditionalAttribute>();
        }
    }
}